src/base/NekRSProblem.C

```diff
@@ -785,37 +792,36 @@ NekRSProblem::addExternalVariables()
 void
 NekRSProblem::calculateMeshVelocity(int e, const field::NekWriteEnum & field)
 {
-  int len = _nek_mesh->nekPolynomialOrder() + 1;
+  int len = _volume? _n_vertices_per_volume : _n_vertices_per_surface;
   double dt = _timestepper->getCurrentDT();
 
+  double * displacement = NULL, *prev_disp = NULL;
```

Comment:
`NULL` should be `nullptr` according to newer C++ standards (`NULL` is discouraged)